#!/usr/bin/env python
PACKAGE = "cloud_factory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#name, paramtype, level, description, default, min, max
gen.add("mode", int_t, 0, "Modes available 0 = thorus, 1 = rectangle", 0, 0, 1)
gen.add("raysIncl", int_t, 0, "vertical inclination rays (layers) = cloud height", 16, 0, 50)
gen.add("inclMin", double_t, 0, "inclMin: lowest inclination angle", -15.0, -45.0, 45.0)
gen.add("inclRes", double_t, 0, "inclRes: resolution of inclination angle", 2.0, 0.1, 4.0)
gen.add("azimRes", double_t, 0, "azimRes: resolution of azimuth angle", 0.2, 0.1, 3.6)
gen.add("azimMin", double_t, 0, "azimMin: (smallest) azimuth start angle", 0.2, 0.1, 3.6)

# gen.add("width", double_t, 0, "Adjust the width of the random cloud", 1.0, 0.1, 10.0)
# gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

#rectang
gen.add("xWidth", double_t, 0, "width of rectangle in x per ray (multiplied by cloud width = raysAzimuth)", 0.01, 0.001, 0.5)
gen.add("yWidth", double_t, 0, "width of rectangle in y per ray (multiplied by cloud width = raysAzimuth)", 0.015, 0.001, 0.5)
gen.add("zHeight", double_t, 0, "height of rectangle in z per ray (multiplied by cloud height = raysIncl)", 0.015, 0.001, 0.5)
gen.add("xStart", double_t, 0, "starting point x for drawing", 0.0, -5.0, 5.0)
gen.add("yStart", double_t, 0, "starting point y for drawing", 2.0, -5.0, 5.0)
gen.add("zStart", double_t, 0, "starting point z for drawing", 0.0, -5.0, 5.0)







exit(gen.generate(PACKAGE, "cloud_factory", "PubShape"))